### DISCUSSION TITLE
orderBy: [ ... ] - sorting by multiple fields fails

### DISCUSSION THREAD

**Discussion by nemanjam** [prisma/prisma#11461] [Q&A] [answered]
Docs says I can sort by multiple fields and I tried that but I get exception.

https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting#sorting

Here is my query:

```typescript
  const users = await prisma.user.findMany({
    ...where,
    skip: (page - 1) * limit,
    take: limit,
    orderBy: [
      { ...(sortDirection && { createdAt: sortDirection as SortDirectionType }) },
      {
        ...(searchTerm && {
          _relevance: {
            fields: ['username', 'name', 'email'],
            search: searchTerm,
            sort: 'desc',
          },
        }),
      },
    ],
  });
```

![image](https://user-images.githubusercontent.com/9990165/151541571-93aa4035-f57a-4047-8e20-0da2fa1e94a3.png)

![image](https://user-images.githubusercontent.com/9990165/151541172-d168c488-a2fc-4311-9b7e-f1d9ddb13bd5.png)

**Comment by austincrim** [comment, marked as answer]
Hey @nemanjam 👋 ,

I believe you are hitting this error because you are passing an empty object as the second element in the array. My query worked locally when removing the empty object.

You will need to conditionally pass the second `orderBy` argument and pass `null` or `undefined` instead of an empty object.

Let me know if I can clarify anything!

**Comment by nemanjam** [comment]
This compiles, thanks.

```typescript
orderBy: [
  ...(!searchTerm
	? [
		{
		  createdAt: sortDirection,
		} as Prisma.UserOrderByWithRelationAndSearchRelevanceInput,
	  ]
	: []),
  ...(searchTerm
	? [
		{
		  _relevance: {
			fields: ['username', 'name', 'email'],
			search: searchTerm,
			sort: 'desc',
		  },
		} as Prisma.UserOrderByWithRelationAndSearchRelevanceInput,
	  ]
	: []),
],
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
